### PR TITLE
VM: Adds -runas flag to qemu

### PR DESCRIPTION
**Pull request by tomponline**
To try and run as either lxd or nobody user.

Signed-off-by: Thomas Parrott <thomas.parrott@canonical.com>

### CODE REVIEW

**Comment by tomponline** [comment]
@stgraber this PR successfully switches the qemu process to run as the lxd user, and the VM starts.

However the 9p mount cannot be established because the config folder on the host is not readable by the lxd user. Would it be OK to change the perms of the config folder and its contents on startup?

**Comment by stgraber** [comment]
@tomponline we'll want the permissions to remain very tight, we absolutely don't want another user on the system being able to read the keys from it. But having some function which recursively changes ownership from root to the temporary user is fine.

**Comment by tomponline** [comment]
@stgraber have updated as discussed thanks.

File: lxd/instance/qemu/vm_qemu.go [hunk 1]
```diff
@@ -634,6 +634,49 @@ func (vm *Qemu) Start(stateful bool) error {
 		"-chroot", vm.Path(),
 	}
 
+	// Detect if it is possible to run as an unprivileged user.
+	var runAsUser string
+	var runAsUID int
+	for _, user := range []string{"lxd", "nobody"} {
```
Comment (stgraber) [inline lxd/instance/qemu/vm_qemu.go]: Should we put some common logic for that somewhere since I believe we're doing the exact same for dnsmasq and probably should do the same for forkdns too.

File: lxd/instance/qemu/vm_qemu.go [hunk 2]
```diff
@@ -1007,8 +1050,9 @@ ConditionPathExists=/dev/virtio-ports/org.linuxcontainers.lxd
 Type=oneshot
 RemainAfterExit=yes
 ExecStartPre=-/sbin/modprobe 9pnet_virtio
-ExecStartPre=/bin/mkdir -p /run/lxd_config
-ExecStart=/bin/mount -t 9p config /run/lxd_config
+ExecStartPre=/bin/mkdir -p /run/lxd_config/9p
+ExecStartPre=/bin/chmod 0700 /run/lxd_config/
+ExecStart=/bin/mount -t 9p config /run/lxd_config/9p -o access=0
```
Comment (stgraber) [inline lxd/instance/qemu/vm_qemu.go]: Can you split the unit changes to a separate commit as that's not directly related.

Comment (stgraber) [reply lxd/instance/qemu/vm_qemu.go]: Done

File: lxd/instance/qemu/vm_qemu.go (hunk 1 above)
Comment (stgraber) [reply lxd/instance/qemu/vm_qemu.go]: Done

**Comment by stgraber** [comment]
@tomponline done the changes I mentioned, will merge once Jenkins is happy
